eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -296,40 +316,73 @@ class NodeRelay private(nodeParams: NodeParams,
+            relayToRecipient(upstream, payloadOut, recipient, paymentCfg, routeParams, useMultiPart = true)
         }
-      case None =>
-        context.log.debug("sending the payment to the next trampoline node")
-        val paymentSecret = randomBytes32() // we generate a new secret to protect against probing attacks
-        val recipient = ClearRecipient(payloadOut.outgoingNodeId, Features.empty, payloadOut.amountToForward, payloadOut.outgoingCltv, paymentSecret, nextTrampolineOnion_opt = Some(packetOut))
-        val payment = SendMultiPartPayment(payFsmAdapters, recipient, nodeParams.maxPaymentAttempts, routeParams)
-        val payFSM = outgoingPaymentFactory.spawnOutgoingPayFSM(context, paymentCfg, multiPart = true)
-        payFSM ! payment
-        payFSM
+      case payloadOut: IntermediatePayload.NodeRelay.ToBlindedPaths =>
+        context.spawnAnonymous(CompactBlindedPathsResolver(router)) ! Resolve(context.messageAdapter[Seq[PaymentBlindedRoute]](WrappedResolvedPaths), payloadOut.outgoingBlindedPaths)
+        waitForResolvedPaths(upstream, payloadOut, paymentCfg, routeParams)
     }
-    payFSM
   }
 
+  private def relayToRecipient(upstream: Upstream.Trampoline,
+                               payloadOut: IntermediatePayload.NodeRelay,
+                               recipient: Recipient,
+                               paymentCfg: SendPaymentConfig,
+                               routeParams: RouteParams,
+                               useMultiPart: Boolean): Behavior[Command] = {
+    val payment =
+      if (useMultiPart) {
+        SendMultiPartPayment(payFsmAdapters, recipient, nodeParams.maxPaymentAttempts, routeParams)
+      } else {
+        SendPaymentToNode(payFsmAdapters, recipient, nodeParams.maxPaymentAttempts, routeParams)
+      }
+    val payFSM = outgoingPaymentFactory.spawnOutgoingPayFSM(context, paymentCfg, useMultiPart)
+    payFSM ! payment
+    sending(upstream, payloadOut, TimestampMilli.now(), fulfilledUpstream = false)
+  }
+
+  /**
+   * Blinded paths in Bolt 12 invoices may encode the introduction node with an scid and a direction: we need to resolve
+   * that to a nodeId in order to reach that introduction node and use the blinded path.
+   */
+  private def waitForResolvedPaths(
+                            upstream: Upstream.Trampoline,
+                            payloadOut: IntermediatePayload.NodeRelay.ToBlindedPaths,
+                            paymentCfg: SendPaymentConfig,
+                            routeParams: RouteParams): Behavior[Command] =
+    Behaviors.receiveMessagePartial {
+      case WrappedResolvedPaths(resolved) if resolved.isEmpty =>
+        context.log.warn(s"rejecting trampoline payment to blinded paths: no usable blinded path")
+        rejectPayment(upstream, Some(TemporaryNodeFailure()))
```

Comment:
`TemporaryNodeFailure` isn't great, we don't have a failure code that perfectly matches this case (we should add some for trampoline), but I think that `UnknownNextPeer` makes a bit more sense here.

Done in #2814